File: sample/src/main/java/contacts/sample/cheatsheet/basics/java/DeleteContactsActivity.java
package contacts.sample.cheatsheet.basics.java;

import static contacts.core.WhereKt.*;

import android.app.Activity;

import java.util.Set;

import contacts.core.ContactsFactory;
import contacts.core.Delete;
import contacts.core.Fields;
import contacts.core.entities.Contact;
import contacts.core.entities.RawContact;

public class DeleteContactsActivity extends Activity {

    Delete.Result deleteContact(Contact contact) {
        return ContactsFactory.create(this)
                .delete()
                .contacts(contact)
                .commit();
    }

    Delete.Result deleteContactWithId(Long contactId) {
        return ContactsFactory.create(this)
                .delete()
                .contactsWithId(contactId)
                .commit();
    }

    Delete.Result deleteNonFavoriteContactsThatHaveANote() {
        return ContactsFactory.create(this)
                .delete()
                .contactsWhereData(
                        and(
                                equalTo(Fields.Contact.Options.Starred, false),
                                isNotNullOrEmpty(Fields.Note.Note)
                        )
                )
                .commit();
    }

    Delete.Result deleteRawContact(RawContact rawContact) {
        return ContactsFactory.create(this)
                .delete()
                .rawContacts(rawContact)
                .commit();
    }

    Delete.Result deleteRawContactWithId(Long rawContactId) {
        return ContactsFactory.create(this)
                .delete()
                .rawContactsWithId(rawContactId)
                .commit();
    }

    Delete.Result deleteRawContactsInTheSetThatHaveANote(Set<Long> rawContactIds) {
        return ContactsFactory.create(this)
                .delete()
                .rawContactsWhereData(
                        and(
                                in(Fields.RawContact.Id, rawContactIds),
                                isNotNullOrEmpty(Fields.Note.Note)
                        )
                )
                .commit();
    }
}

File: sample/src/main/java/contacts/sample/cheatsheet/basics/kotlin/DeleteContactsActivity.kt
package contacts.sample.cheatsheet.basics.kotlin

import android.app.Activity
import contacts.core.*
import contacts.core.entities.Contact
import contacts.core.entities.RawContact

class DeleteContactsActivity : Activity() {

    fun deleteContact(contact: Contact): Delete.Result = Contacts(this)
        .delete()
        .contacts(contact)
        .commit()

    fun deleteContactWithId(contactId: Long): Delete.Result = Contacts(this)
        .delete()
        .contactsWithId(contactId)
        .commit()

    fun deleteNonFavoriteContactsThatHaveANote(): Delete.Result = Contacts(this)
        .delete()
        .contactsWhereData {
            (Contact.Options.Starred equalTo false) and Note.Note.isNotNullOrEmpty()
        }
        .commit()

    fun deleteRawContact(rawContact: RawContact): Delete.Result = Contacts(this)
        .delete()
        .rawContacts(rawContact)
        .commit()

    fun deleteRawContactWithId(rawContactId: Long): Delete.Result = Contacts(this)
        .delete()
        .rawContactsWithId(rawContactId)
        .commit()

    fun deleteRawContactsInTheSetThatHaveANote(rawContactIds: Set<Long>): Delete.Result =
        Contacts(this)
            .delete()
            .rawContactsWhereData {
                (RawContact.Id `in` rawContactIds) and Note.Note.isNotNullOrEmpty()
            }
            .commit()
}